src/Dashboard/Grid.php

```diff
@@ -1097,6 +1097,31 @@ protected function getMenuItemtypes(): array
         return $menu_itemtypes;
     }
 
+    /**
+     * Get cache key for the "getAllDasboardCards" function data.
+     * The data will contain some translated strings and thus must be kept in a
+     * separate cache entry for each languages
+     *
+     * @return string
+     */
+    public static function getAllDashboardCardsCacheKey(): string
+    {
+        $language = Session::getLanguage();
+
+        if (!$language) {
+            // Keep a log trace in case of unexpected session value
+            // This may help with debugging in the future
+            trigger_error(
+                "Unable to get current language, will default to 'getAllDasboardCards' key",
+                E_USER_WARNING
+            );
+
+            return "getAllDasboardCards";
+        }
+
+        return "getAllDasboardCards_$language";
```

Comment:
We can fix the cache key typo that was already present.

```suggestion
            return "getAllDashboardCards";
        }

        return "getAllDashboardCards_$language";
```